google/cloud/storage/blob.py

```diff
@@ -4700,6 +4708,32 @@ def retention(self):
         info = self._properties.get("retention", {})
         return Retention.from_api_repr(info, self)
 
+    @property
+    def soft_delete_time(self):
+        """If this object has been soft-deleted, returns the time at which it became soft-deleted.
+
+        :rtype: :class:`datetime.datetime` or ``NoneType``
+        :returns:
+            (readonly) The time that the object became soft-deleted.
+             Note this property is only set for soft-deleted objects.
+        """
+        soft_delete_time = self._properties.get("softDeleteTime")
+        if soft_delete_time is not None:
+            return _rfc3339_nanos_to_datetime(soft_delete_time)
+
+    @property
+    def hard_delete_time(self):
```

Comment:
I'm kind of tempted to call this something like soft_delete_expiration_time. It seems a bit confusing to call it hard_delete_time, since it's such a different concept from the mirrored soft_delete_time. Let's discuss.

Reply:
I understand where you're coming from. However `hardDeleteTime` is the official object property name, and I think it's best that we stay consistent with the API and across languages for a better developer experience. reference internal doc go/gcs-soft-delete-user-guide